src/Data/Textile/KnittingProcess.elm

```diff
@@ -0,0 +1,91 @@
+module Data.Textile.KnittingProcess exposing
+    ( KnittingProcess(..)
+    , decode
+    , encode
+    , fromString
+    , toLabel
+    , toString
+    )
+
+import Json.Decode as Decode exposing (Decoder)
+import Json.Decode.Extra as DE
+import Json.Encode as Encode
+
+
+type KnittingProcess
+    = Mix
+    | FullyFashioned
+    | Seamless
+    | Circular
+    | Straight
+
+
+decode : Decoder KnittingProcess
+decode =
+    Decode.string
+        |> Decode.andThen (fromString >> DE.fromResult)
+
+
+encode : KnittingProcess -> Encode.Value
+encode =
+    toString >> Encode.string
+
+
+fromString : String -> Result String KnittingProcess
+fromString string =
+    case string of
+        "knitting-mix" ->
+            Ok Mix
+
+        "knitting-fully-fashioned" ->
+            Ok FullyFashioned
+
+        "knitting-seamless" ->
+            Ok Seamless
+
+        "knitting-circular" ->
+            Ok Circular
+
+        "knitting-straight" ->
+            Ok Straight
```

Comment:
I'm not sure the `knitting-` prefix adds a lot of value; what would be the case where it's important or useful?